game-engine/domain/src/main/java/za/co/entelect/challenge/entities/Building.java

```diff
@@ -126,11 +126,11 @@ public void damageSelfDirectly(int damageTaken, TowerDefensePlayer missileOwner)
     }
 
     private void setHealthAndAddScore(int damageTaken, TowerDefensePlayer missileOwner) {
-
+        int previousHealth = health;
         health -= damageTaken;
         health = Math.max(0, health);
 
-        missileOwner.addScore(damageTaken * destroyMultiplier);
+        missileOwner.addScore((previousHealth - health) * destroyMultiplier);
```

Comment:
damageTaken was always the weapons full damage...but 1hp buildings should not give 20dmg score